patchwerk-sim.py

```diff
@@ -74,42 +74,95 @@ def __str__(self):
+        self.name = name
+        self.max_health = max_health
+        self.dodge_parry = dodge_parry
+        self.mitigation = mitigation
+        self.current_health = self.max_health
+
+    def reset(self):
+        self.current_health = self.max_health
+
+    # returns True if tank dies, False otherwise
+    def get_smashed(self, dmg):
+        # check for misses
+        if random.random() < self.dodge_parry:
+            # print("Hateful Strike MISSES {}".format(self.name))
+            return False
+
+        # print("Hateful Strike hits {} ({} hp) for {} dmg".format(self.name, self.current_health, dmg))
+        self.current_health -= dmg
+        if self.current_health <= 0:
+            # print("{} has DIED! ({} Overkill)".format(self.name, -self.current_health))
+            return True
+        return False
+
+    # returns a tuple with total raw healing and overhealing
+    def get_healed(self, heal_qty):
+        # print("{} ({} hp) is healed for {}".format(self.name, self.current_health, heal_qty))
+        self.current_health += heal_qty
+        overhealing = 0
+        if self.current_health > self.max_health:
+            overhealing = self.current_health - self.max_health
+            self.current_health = self.max_health
+
+        return (heal_qty, overhealing)
+
+    def __str__(self):
+        return self.name
+
 # updated hateful strike to hit every 1.2s instead of random number from 1.2 to 2s
 def get_timetonext_hateful():
     return 1.2
-    # seconds_later = (random.random() * (0.8499) + 1.2)
-#     return round(seconds_later, 1)
 
-def get_hateful_target(tanks_health):
-    return tanks_health.index(max(tanks_health))
+# returns a tuple of (tank_to_be_hit_next, tank_index)
+def get_hateful_target(tanks):
+    highest_health_tank_index = -1
+    highest_health_tank = -999
+
+    for index, tank in enumerate(tanks):
+        if tank.current_health > highest_health_tank:
+            highest_health_tank = tank.current_health
+            highest_health_tank_index = index
+
+    return tanks[highest_health_tank_index], highest_health_tank_index
+
+# def get_hateful_target(tanks_health):
+#     return tanks_health.index(max(tanks_health))
 
 # tank 0 is healed by healers [1, 2, 3], tank 1 by healers [4, 5, 6], tank 2 by healers [7, 8, 9]
 def get_heal_target(healer_idx):
     return (healer_idx + 2) // 3 - 1
 
-# returns a tuple with total raw healing and overhealing
-def heal_tank(tanks_health, tank_idx, heal_qty, max_tanks_health):
-    # print("Tank #{} ({} hp) is healed for {}".format(tank_idx, tanks_health[tank_idx], heal_qty))
-    tanks_health[tank_idx] += heal_qty
-    overhealing = 0
-    if tanks_health[tank_idx] > max_tanks_health:
-        overhealing = tanks_health[tank_idx] - max_tanks_health
-        tanks_health[tank_idx] = max_tanks_health
-
-    return (heal_qty, overhealing)
+# # returns a tuple with total raw healing and overhealing
```

Comment:
i think we can remove the code you replaced, git will track it in its history if we ever need to refer to it